### ISSUE TITLE
false negative on clippy::filter_next when filter and next wraps map

### ISSUE THREAD

**Issue by meltinglava** [rust-lang/rust-clippy#6423]
False negative when filter and next is wrapped around a map.

There might be a few other methods like map that will have the same effect.

I tried this code:

```rust
    v.windows(3)
        .enumerate()
        .filter(|(_, a)| a == &[true, false, true])
        .map(|(v, _)| v + 1)
        .next()
        .unwrap()
```

I expected to see this happen: Clippy to suggest
```rs
    v.windows(3)
        .enumerate()
        .find(|(_, a)| a == &[true, false, true])
        .map(|(v, _)| v + 1)
        .unwrap()
```
Instead, this happened: Nothing

### Meta

- `cargo clippy -V`: e.g. clippy 0.0.212 (1c389ff 2020-11-24)rust
- `rustc -Vv`:
  ```
  rustc 1.50.0-nightly (1c389ffef 2020-11-24)
  binary: rustc
  commit-hash: 1c389ffeff814726dec325f0f2b0c99107df2673
  commit-date: 2020-11-24
  host: x86_64-unknown-linux-gnu
  release: 1.50.0-nightly
  ```

**Comment by flip1995** [comment]
We could even suggest using `find_map` here.

**Comment by meltinglava** [comment]
> 
> 
> We could even suggest using `find_map` here.

#4193 will have the same prolem here. I think its always safe to suggest find, and when the conservativness of find_map is figured out it can be applied here.

**Comment by mi2ebi** [comment]
bumped into this today, a fix would be appreciated :3
